### ISSUE TITLE
Antialiasing issues

### ISSUE THREAD

**Issue by 9inpachi** [HSF/phoenix#101]
Hi,

After some debugging, I rendered the main renderer after the `EffectComposer` renders and apparently antialiasing worked. The antialiasing issue CAN BE solved but there are some problems.
I found that antialiasing doesn't work with `EffectComposer` (see [THIS](https://github.com/mrdoob/three.js/issues/4287#issuecomment-43917659)) or post processing in general because of the limitation in WebGL. Now if we don't render the `composer` (see [HERE](https://github.com/HSF/phoenix/blob/master/src/app/services/three/selection-manager.ts#L78)) after the main renderer we can't have the `OutlinePass` work which affects selection.

Apart from this, antialiasing brings a drop of about 10 FPS. We might be able to make use of the `FXAAShader` for an increase in quality as an alternative to native antialiasing.

**Comment by EdwardMoyse** [comment]
Ouch - this is painful. So we either give up on the nice glow effect, or we cannot have AA. :-(

This might be another thing we should make switchable. Another option for selection would be to change the colour of the objects in question...

**Comment by 9inpachi** [comment]
@EdwardMoyse 

In order to make it switchable we have to have a separate `WebGLRenderer` because according to #8265 dynamically setting the antialiasing is not possible.

The solution I have is to have antialias enabled by default. Now if the `EffectComposer` renders the scene, this antialias will automatically not work (hence disabled). And if we want to toggle antialiasing through a button we will not render the composer and else call the render function of the renderer directly. That is we can use the `EffectComposer` as a secondary renderer when antialiasing is to be disabled.

Like this.

```ts
public render(camera: Camera, scene: Scene) {
  if (this.composer && !this.antialiasing) {
    this.defaultRenderPass.camera = camera;
    this.defaultRenderPass.scene = scene;
    this.composer.render();
  } else if (this.antialiasing) {
    this.composer.renderer.render(scene, camera);
  }
}
```

One other solution is to have antialias on by default and disable antialiasing on selection mode.

TL;DR Do we want a toggle for antialiasing or just disable antialiasing on selection?

**Comment by EdwardMoyse** [comment]
Are you proposing that antialiasing would automatically disable itself if a user selects the pick icon? That could be worth a try ...

**Comment by 9inpachi** [comment]
> Are you proposing that antialiasing would automatically disable itself if a user selects the pick icon?

Yes it will. The other option is to have a performance toggle which can enable disable the antialiasing manually. And always set the antialiasing to be false in selection.

**Comment by EdwardMoyse** [comment]
I think in general we should probably aim to have some more options/toggles, but this sounds like a good solution for now.

**Comment by 9inpachi** [comment]
Done with #180

@EdwardMoyse if you have any suggestions or something related to this issue, you can reopen it.
